### ISSUE TITLE
[Bug]: Videos do not properly fullscreen with fractional scaling.

### ISSUE THREAD

**Issue by ShinyGil** [Floorp-Projects/Floorp#1298]
### Pre-Submission Checklist

- [X] I have checked the existing/closed issues to ensure this issue has not already been reported.
- [X] I am using the latest supported version of the browser.
- [X] I have tried disabling all browser extensions/add-ons to rule out interference from these.
- [X] I have tried clearing the browser cache and cookies.
- [X] I have checked with a new profile and the issue still occurs.
- [X] I have provided detailed step-by-step instructions on how to reproduce the issue.
- [X] I have included relevant screenshots or console outputs.
- [X] I have checked that this problem is not occurring in the latest version of Firefox ESR.
- [X] This issue is specific to this browser and does not occur in other browsers.

### Bug Summary

When using Floorp with my display scaled at 125%, when I fullscreen a video, instead of fullscreening the video instead goes into the corner at the same size and the rest of the screen becomes grey. Does not occur on 100% scaling. Firefox ESR 115.13 does not have the issue on 125%.

### Reproduction Steps

1. Set display to 125%
2. Go to YouTube
3. Open a video
4. Fullscreen the video

### Expected Behavior

The video becomes fullscreen as it would in other browsers, with no grey borders.

### Actual Behavior

Video does not fullscreen and instead goes into the top left corner, surrounded by grey borders.

### Screenshots

![Screenshot_20240731_230747](https://github.com/user-attachments/assets/28d258fb-1f13-4f61-a43e-75790e86d27a)

### Browser Version

 11.18.1

### Operating System

Fedora 40, EndeavourOS (Arch)

### Additional Context

This was the output from running Floorp in a terminal window:
```console.warn: "CSK Init Completed"
console.log: "add window"
console.error: ({})
console.log: ({initialized:true})
JavaScript error: resource://gre/modules/XULStore.sys.mjs, line 60: Error: Can't find profile directory.
JavaScript warning: https://www.google.com/js/th/wQkdmVX08K_HD_9NHn0QjGfgu04J0GwVvAmYELAxlpM.js line 2 > eval line 7097 > eval line 1 > eval line 1 > eval, line 1: WEBGL_debug_renderer_info is deprecated in Firefox and will be removed. Please use RENDERER.

**Comment by ShinyGil** [comment]
Commenting simply to add the issue is still present in Floorp 11.16.

**Comment by surapunoyousei** [comment]
now, Floorp is based on ESR128. This issue maybe fixed

**Comment by ShinyGil** [comment]
> now, Floorp is based on ESR128. This issue maybe fixed

Floorp 11.17.3, issue still persists unfortunately.

**Comment by ShinyGil** [comment]
> bug is still present 11.18.1

I can also confirm this.

**Comment by ShinyGil** [comment]
> bug is still present 11.18.1

I can also confirm this.

@surapunoyousei This issue is still present on ll versions since the issue was made, and I have also confirmed it happens on Arch as well. I have updated the issue post to reflect the new version of Floorp and the other OS I have seen this behavior on.

**Comment by surapunoyousei** [comment]
I don't use Linux so I can't fix this, but would welcome a patch!

**Comment by Thesola10** [comment]
Have you tried setting `widget.wayland.fractional-scale.enabled` to `false` to use downscaled integer scaling? This could help narrow the issue down to the Wayland fractional scale protocol.

**Comment by ShinyGil** [comment]
> Have you tried setting `widget.wayland.fractional-scale.enabled` to `false` to use downscaled integer scaling? This could help narrow the issue down to the Wayland fractional scale protocol.

It is on false.
![image](https://github.com/user-attachments/assets/acacf680-a14a-40d1-b990-4ca603287184)

**Comment by ShinyGil** [comment]
@Thesola10 You were onto something though, setting it to `true` solves it! Previously it didn't when I tried so I had set it back to `false`.

**Comment by ShinyGil** [comment]
> @Thesola10 You were onto something though, setting it to `true` solves it! Previously it didn't when I tried so I had set it back to `false`.

New issue however...enabling that flag makes the right click context menu unusable. 
![image](https://github.com/user-attachments/assets/44771cc9-c4b2-4834-8074-7c3ab9ebbdaf)

**Comment by Sridou** [comment]
For what it's worth, I can confirm that this problem only occurs when using Lepton UI mode. It works as intended in default Firefox Proton UI and Floorp Fluerial UI.  I can also confirm that using the official Lepton UI in Vanilla Firefox using their arch install script causes the same issue when fullscreening in fullscreen mode [F11] . I don't know how to cross reference issues in Github, as still newbie [I'll learn soon], so it would be helpful if someone could  follow on this.

**Comment by ShinyGil** [comment]
> For what it's worth, I can confirm that this problem only occurs when using Lepton UI mode. It works as intended in default Firefox Proton UI and Floorp Fluerial UI. I can also confirm that using the official Lepton UI in Vanilla Firefox using their arch install script causes the same issue when fullscreening in fullscreen mode [F11] . I don't know how to cross reference issues in Github, as still newbie [I'll learn soon], so it would be helpful if someone could follow on this.

You're right! It's Lepton! Agh, that is such a shame, Lepton is my favorite. If no one has opened an issue with them, I will as soon as possible.
